### ISSUE TITLE
Goto Definition does not find definitions of symbols outside the workspace folder by default

### ISSUE THREAD

**Issue by KaraRyougi** [microsoft/vscode-cpptools#2053]
macOS High Sierra 10.13.4
    VSCode 1.23.1
    C/C++ Extension 0.17.3

### How to reproduce this bug:

1. Organize include file as follows:

```
.
└── Components
    ├── header.c
    └── include
        └── header.h
```

The header contains a function `includeTest()`

2. Config includePath:
```
"...my path.../Components/**"
```

3. Create a new file in other workspace:
```
#include <header.h>

int main() {
    return includeTest();
}
```

4. Right-click on the function name `includeTest()` in the newly created file and select `Go to Definition`.

5. VSCode will jump to `header.h` instead of `header.c`.

**Comment by bobbrow** [comment]
go to definition uses the `"browse.path"` property.  Is `..my path../Components` in the `browse.path`?  If not, then the code navigation service is likely returning the declaration because it doesn't know where the definition is.

**Comment by KaraRyougi** [comment]
Thanks! It turns out I didn't set `browse.path` correctly.

**Comment by KaraRyougi** [comment]
I set `browse.path` the same as `includePath`, but go to definition still doesn't work as expected.
Configuration file:

```
...
            "browse": {
                "path": [
                    "${workspaceFolder}",
                    "...my path.../Components/**"
                ],
                "limitSymbolsToIncludedHeaders": true
            },
            "includePath": [
                "${workspaceFolder}/**",
                "...my path.../Components/**"
            ],
...
```

**Comment by bobbrow** [comment]
Oh... I think I know what's going on.  Your Components folder isn't under the ${workspaceFolder} path is it?

By default, the extension doesn't parse files outside the ${workspaceFolder} unless they are `#include`ed.  .c/.cpp files are not usually included by other files, which means that we will likely only have the declarations available by default.  The `"limitSymbolsToIncludedHeaders"` property can toggle this behavior and will tell the extension to parse everything in your `browse.path` instead of waiting until you open those files to parse them.

So you have two options:
1. Set `limitSymbolsToIncludedHeaders` to `false` and have the extension parse everything in the `browse.path` (this can take a while, but should only be a one-time cost)
2. Manually open all of the .c/.cpp files under the Components folder in VS Code to force them to be parsed (doesn't scale well if you have a lot of files)

@sean-mcmanus, we should revisit this decision now that the extension has a better separation of system includes from project includes.  This property was intended to reduce the parsing load incurred by the system headers, but I think we need to allow the extension to parse through other libraries outside the workspace folder that are not header-only.

**Comment by sean-mcmanus** [comment]
@bobbrow Hmm...but there might still be perf issues if users use some giant library like boost? Not sure, but we could still add a setting to make it configurable.

**Comment by bobbrow** [comment]
I'm thinking we would only parse .c/.cpp files outside the workspace folder.  Headers can continue to respect the `limitSymbolsToIncludedHeaders` property.

**Comment by sean-mcmanus** [comment]
@bobbrow Yeah, that makes sense...not sure why we didn't do that originally...should only require a small change to check the file extension.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.17.5.
